include/deal.II/non_matching/mapping_info.h

```diff
@@ -157,30 +293,550 @@ namespace NonMatching
   void
   MappingInfo<dim, spacedim>::reinit(
     const typename Triangulation<dim, spacedim>::cell_iterator &cell,
-    const ArrayView<const Point<dim>> &                         unit_points)
+    const ArrayView<const Point<dim>> &                         unit_points_in)
+  {
+    unit_points =
+      std::vector<Point<dim>>(unit_points_in.begin(), unit_points_in.end());
+
+    mapping_data.resize(1);
+    compute_mapping_data_for_generic_points(cell,
+                                            unit_points_in,
+                                            mapping_data[0]);
+
+    state = State::single_cell;
+  }
+
+
+  template <int dim, int spacedim>
+  void
+  MappingInfo<dim, spacedim>::reinit(
+    const typename Triangulation<dim, spacedim>::cell_iterator &cell,
+    const Quadrature<dim> &                                     quadrature)
+  {
+    const auto &points  = quadrature.get_points();
+    const auto &weights = quadrature.get_weights();
+
+    reinit(cell, points);
+
+    if (update_flags_mapping & update_JxW_values)
+      for (unsigned int q = 0; q < points.size(); ++q)
+        mapping_data[0].JxW_values[q] =
+          determinant(Tensor<2, dim>(mapping_data[0].jacobians[q])) *
+          weights[q];
+  }
+
+
+  template <int dim, int spacedim>
+  template <typename Iterator>
+  void
+  MappingInfo<dim, spacedim>::reinit_cells(
+    const IteratorRange<Iterator> &             cell_iterator_range,
+    const std::vector<std::vector<Point<dim>>> &unit_points_vector,
+    const unsigned int                          n_unfiltered_cells)
+  {
+    do_cell_index_compression =
+      n_unfiltered_cells != numbers::invalid_unsigned_int;
+
+    const unsigned int n_cells =
+      std::distance(cell_iterator_range.begin(), cell_iterator_range.end());
+
+    Assert(n_cells == unit_points_vector.size(),
+           ExcDimensionMismatch(n_cells, unit_points_vector.size()));
```

Comment:
Can you please switch the assignments around?
```suggestion
    const unsigned int n_cells = unit_points_vector.size();

    Assert(n_cells == std::distance(cell_iterator_range.begin(), cell_iterator_range.end()),
           ExcDimensionMismatch(n_cells, unit_points_vector.size()));
```
The reason is that `std::distance` might need to loop through all entries, which should be avoided in release mode. It is enough to do it once further down.

Reply:
I have replaced the `Assert` through `AssertDimension` to make sure `std::distance` is only called when necessary and only once in debug mode.